### ISSUE TITLE
warning: filtered namespace in public API

### ISSUE THREAD

**Issue by vinniefalco** [cppalliance/mrdox#337]
When a symbol that would be otherwise skipped by the namespace filter appears anyway because it is part of another public API, a warning should be emitted:

```
namespace detail {
struct my_shame { };
}

struct T : detail::my_shame { };
```

Should produce the message

```
warning: filtered symbol detail::my_shame in public API at "source.cpp" line 42
```

This can apply to return types, function arguments, and template parameters as well (and probably other places) for example:
```
detail::my_shame publicFunction();
```

**Comment by klemens-morgenstern** [comment]
Shouldn't this be option? E.g. *unspecified* or sth? This is very common in asio.

```cpp
struct socket
{
   using native_type = detail::something; // 
};
```

where `native_type` becomes _implementation_defined_.

See below.

https://www.boost.org/doc/libs/1_80_0/doc/html/boost_asio/reference/basic_socket/native_handle_type.html

**Comment by vinniefalco** [comment]
Can you please link the asio source code for this?

**Comment by klemens-morgenstern** [comment]
It's here however: 

https://www.boost.org/doc/libs/1_82_0/doc/html/boost/process/basic_pipe.html

https://github.com/boostorg/process/blob/develop/include/boost/process/pipe.hpp#L44

**Comment by vinniefalco** [comment]
Yes we need a way to do that but probably not tie it directly to any `detail` namespace token, although we could have some kind of option for that perhaps.
